frontend/src/globals/unity-util.ts

```diff
@@ -2111,15 +2111,18 @@ export class UnityUtil {
 	 * A helper function to split the calls into multiple calls when the array is too large for SendMessage to handle
 	 */
 	public static multipleCallInChunks(arrLength: number, func:(start: number, end: number) => any, chunkSize = 5000) {
-		let index = 0;
-		while (index < arrLength) {
-			const end = index + chunkSize >= arrLength ? undefined : index + chunkSize;
-			const i = index; // For the closure
-			this.unityOnUpdateActions.push(() => {
-				func(i, end);
-			});
-			index += chunkSize;
-		}
+		return new Promise((resolve) => {
+			let index = 0;
+			while (index < arrLength) {
+				const end = index + chunkSize >= arrLength ? undefined : index + chunkSize;
+				const i = index; // For the closure
+				this.unityOnUpdateActions.push(() => {
+					func(i, end);
+				});
+				index += chunkSize;
+			}
+			this.unityOnUpdateActions.push(resolve);
```

Comment:
@sebjf  can you check if the change in this loop makes any sense?

Reply:
Hi @carmenfan the loop looks sensible to me, and should do what was discussed in the thread before (resolve a promise when all the chunks have been sent to unity (and its up to the caller to interpret the significance of that, based on whether the calls inside unity are synchronous or not).